daml-assistant/src/DAML/Assistant/Version.hs

```diff
@@ -50,20 +50,38 @@ getAssistantSdkVersion = do
 -- AssistantError exception if the version cannot be determined.
 getSdkVersionFromSdkPath :: SdkPath -> IO SdkVersion
 getSdkVersionFromSdkPath sdkPath = do
-    config <- requiredIO "Failed to read SDK config." $
+    config <- requiredAny "Failed to read SDK config." $
         readSdkConfig sdkPath
     requiredE "Failed to parse SDK version from SDK config." $
         sdkVersionFromSdkConfig config
 
 -- | Determine SDK version from project root. Fails with an
 -- AssistantError exception if the version cannot be determined.
 getSdkVersionFromProjectPath :: ProjectPath -> IO SdkVersion
-getSdkVersionFromProjectPath projectPath = do
-    config <- requiredIO "Failed to read project config." $
-        readProjectConfig projectPath
-    versionM <- requiredE "Failed to parse SDK version from project config." $
-        sdkVersionFromProjectConfig config
-    required "SDK version missing from project config." versionM
+getSdkVersionFromProjectPath projectPath =
+    requiredIO ("Failed to read SDK version from " <> pack projectConfigName) $ do
+        configE <- tryConfig $ readProjectConfig projectPath
+        case sdkVersionFromProjectConfig =<< configE of
+            Right (Just v) ->
+                pure v
+            Left (ConfigFileInvalid _ raw) ->
+                throwIO $ assistantErrorDetails
+                    (projectConfigName <> " is an invalid YAML file")
+                    [("path", unwrapProjectPath projectPath </> projectConfigName)
+                    ,("internal", displayException raw)]
+            Right Nothing ->
+                throwIO $ assistantErrorDetails
+                    ("sdk-version field is missing from " <> projectConfigName)
+                    [("path", unwrapProjectPath projectPath </> projectConfigName)]
+            Left (ConfigFieldMissing _ _) ->
+                throwIO $ assistantErrorDetails
+                    ("sdk-version field is missing from " <> projectConfigName)
+                    [("path", unwrapProjectPath projectPath </> projectConfigName)]
+            Left (ConfigFieldInvalid _ _ raw) ->
+                throwIO $ assistantErrorDetails
+                    ("sdk-version field is invalid in " <> projectConfigName)
+                    [("path", unwrapProjectPath projectPath </> projectConfigName)
+                    ,("internal", raw)]
```

Comment:
Do you need to `displayException raw` here too?

Reply:
Nope! It's a string. It's a little confusing but these are the return types of the errors we get from Data.Yaml...